File: dotnet/UrlShortener/Controllers/EncodeController.cs
using Microsoft.AspNetCore.Mvc;
using UrlShortener.Models;

namespace UrlShortener.Controllers;

[ApiController]
[Route("[controller]")]
public class EncodeController : ControllerBase
{
    private readonly ILogger<EncodeController> _logger;

    public EncodeController(ILogger<EncodeController> logger)
    {
        _logger = logger;
    }

    [HttpPost]
    public IEnumerable<EncodeResponse> Post(EncodeRequest encodeRequest) => throw new NotImplementedException();
}

File: nestjs/src/controllers/encode.controller.ts
import { Controller, NotImplementedException, Post, Body } from '@nestjs/common';
import { AppService } from '../services/app.service';
import { EncodeRequest } from '../models/EncodeRequest';
import { EncodeResponse } from '../models/EncodeResponse';

@Controller()
export class EncodeController {
  constructor(private readonly appService: AppService) {}

  @Post('/encode')
  encode(@Body() encodeReqeust: EncodeRequest): EncodeResponse {
    throw new NotImplementedException
  }
}
